test/test_data.py

```diff
@@ -1,13 +1,90 @@
+from typing import Union, Dict, List, Tuple
 import unittest
+
+from rdflib.util import from_n3
+from rdflib import Graph, URIRef, Literal, BNode
+import pytest
+
 from pyontutils.core import OntResIri
 
+Semantic = Union[URIRef, Literal, BNode]
+
 
 class TestCurationExportTtl(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         cls.ori = OntResIri('https://cassava.ucsd.edu/sparc/exports/curation-export.ttl')
         cls.graph = cls.ori.graph
+        cls.spaql_templates = SparqlQueryTemplates(cls.ori.graph.namespace_manager)
 
-    def test(self):
+    def test_dataset_group(self):
         """ sparql queries here """
-        self.graph
+        query = self.spaql_templates.dataset_group('https://api.blackfynn.io/datasets/N:dataset:c2a014b8-2c15-4269-b10a-3345420e3d56/subjects/53')
+        assert len(list(self.graph.query(query))) > 0
+
+    def test_related_datasets(self):
+        query = self.spaql_templates.related_datasets('dataset:bec4d335-9377-4863-9017-ecd01170f354')
+        assert len(list(self.graph.query(query))) > 0
+
+
+class SparqlQueryTemplates:
+    """ Creates SPARQL query templates. """
+
+    def __init__(self, nsm=None):
+        self.nsm = nsm if nsm else Graph().namespace_manager
+        self.prefixes = {p:ns for p, ns in self.nsm.namespaces() if p}
+
+    def sparql_uri(self, uri_or_curie: Union[URIRef, str]) -> str:
+        """ Converts IRIs and curies to a usable format for SPARQL queries. """
+        uri_or_curie = str(uri_or_curie)
+        if uri_or_curie.startswith('http') or isinstance(uri_or_curie, URIRef):
+            return '<'+str(uri_or_curie)+'>'
+        try:
+            return '<'+str(from_n3(uri_or_curie, nsm=self.nsm))+'>'
+        except:
+            ValueError(f'Invalid prefix used {uri_or_curie}')
+
+    def prefixes_used(self, query: str) -> str:
```

Comment:
Does rdflib have a native sparql parser that provides this functionality?

Reply:
Not that I'm aware of. It if you give iri without carrots it fails. I'll search if there is one hiding.

Reply:
It looks like what we actually want in this situation is `initNs` re https://rdflib.readthedocs.io/en/stable/intro_to_sparql.html#prepared-queries.

Reply:
Oh awesome! I just went ahead and dumped all prefixes since it's all tucked away anyway. I had to "from rdflib.plugins import sparql" though since plugins oddly doesn't exist from straight from an rdflib import. I don't know if that was intended or that is a bug.